Procedural Dungeon Generator

```diff
@@ -1,29 +1,43 @@
-    This will act as a step-by-step checklist to help people build unique maps.
+    This will act as a step-by-step list of guidelines to help people build fun, functioning 
+maps. It may be a good idea to have a few pages of notes detailing any special rules 
+for your map, as well as the inventories, gear, and personalities of your characters.
 
 
 ==============================
 The Base Map
 ==============================
     
         Firstly, determine the planet or region this map will take place in, as 
-well as any area-specific rules that may apply. Some examples include the map 
-being on a water world with most or all of the environment underwater, in space 
-with no air or gravity, in a snowstorm with very limited visibility but also the 
-ability to track people based on their footprints, on a volcanic planet where 
-staying outside too long without a mask can injure or blind the players, etc. 
-If no specific rules exist for the map you want to make, it will be up to you to 
-create them. In general, map-specific rules should not be a major hinderance to 
-the players, but should make the experience more interesting and challenging. If 
-traversing the map is harder than fighting the enemies, then it won't be a fun game
-for anybody.
+well as any area-specific rules that may apply. 
+
+    Some examples include:
+    * The map being on a water world with most or all of the environment underwater. The 
+characters would either have to use underwater breathing gear, or the map-builder would 
+have to write their own breath-holding rules.
+    * In space with no air or gravity. There are rules in place dealing with movement in 
+microgravity, as well as EVA suits that the players could use. 
+    * In a snowstorm with very limited visibility but also the ability to track people based on 
+their footprints. The hunter class might be very useful in a situation such as this, because 
```

Comment:
What you're driving at here is that a good dungeon has something for every class to excel at. We should say that at some point.